### ISSUE TITLE
Solve: The system don't is solved when we request one symbol with one depend of eq

### ISSUE THREAD

**Issue by latot** [sympy/sympy#11655]
Hi, this time its the system don't is simplified when we request one symbol, and it depends only of one eq:
```
>>> solve([Eq(y,1), Eq(x+y,1)], x)
{x: -y + 1}
```

Thx. Cya.

**Comment by Shekharrajak** [comment]
`solve` or `solveset` do nothing, if in the equation no `symbol` is present.

```
In [45]: substitution([Eq(y,1), Eq(x+y,1)], [x, y])
Out[45]: {(0, 1)}

In [46]: nonlinsolve([Eq(y,1), Eq(x+y,1)], [x, y])
Out[46]: {(0, 1)}

In [47]: solve([Eq(y,1), Eq(x+y,1)], [x, y])
Out[47]: {x: 0, y: 1}
```

**Comment by latot** [comment]
Yes, but the problem is the system don't is solved when we request the var, we don't always need al the system like in simplex, and in this case the symbol exist, but only in one eq.

**Comment by Shekharrajak** [comment]
E.g.

```
In [14]: solve([Eq(z + y,1), Eq(x,1)], x)
Out[14]: {x: 1}
```

So, it should not solve for other symbols (`y, z`) , because user is not interested on that. Right ?

**Comment by latot** [comment]
m, you are right, let me update the tittle for the condition, when depends of one eq in a related system.

**Comment by latot** [comment]
mm, i'll explain a little this, when i refer to a related system its about the requested var, with the actual examples:
```
In [14]: solve([Eq(z + y,1), Eq(x,1)], x)
Out[14]: {x: 1}
```
here we have a system, but we only one eq for the system of x, in this similar case:
```
>>> solve([Eq(y,1), Eq(x+y,1)], x)
{x: -y + 1}
```
we have a system, but the system of x have two eq, so its a related system on x.

Thx. Cya.
